tests/modules/internalsecret.c

```diff
@@ -0,0 +1,152 @@
+#include "redismodule.h"
+#include <errno.h>
+
+int InternalAuth_GetInternalSecret(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    REDISMODULE_NOT_USED(argv);
+    REDISMODULE_NOT_USED(argc);
+
+    size_t len;
+    const char *secret = RedisModule_GetInternalSecret(ctx, &len);
+    RedisModule_ReplyWithStringBuffer(ctx, secret, len);
+    return REDISMODULE_OK;
+}
+
+int InternalAuth_InternalCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    REDISMODULE_NOT_USED(argv);
+    REDISMODULE_NOT_USED(argc);
+
+    RedisModule_ReplyWithSimpleString(ctx, "OK");
+    return REDISMODULE_OK;
+}
+
+/* A wrap for RM_Call that pass the 'C' flag to pass the current client. */
+int rm_call_withclient(RedisModuleCtx *ctx, RedisModuleString **argv, int argc){
+    if(argc < 2){
+        return RedisModule_WrongArity(ctx);
+    }
+
+    const char* cmd = RedisModule_StringPtrLen(argv[1], NULL);
+
+    RedisModuleCallReply* rep = RedisModule_Call(ctx, cmd, "vC", argv + 2, argc - 2);
+    if(!rep) {
+        char err[100];
+        switch (errno) {
+            case EACCES:
+                RedisModule_ReplyWithError(ctx, "ERR NOPERM");
+                break;
+            case ENOENT:
+                RedisModule_ReplyWithError(ctx, "ERR unknown command");
+            default:
+                snprintf(err, sizeof(err) - 1, "ERR errno=%d", errno);
+                RedisModule_ReplyWithError(ctx, err);
+                break;
+        }
+    } else {
+        RedisModule_ReplyWithCallReply(ctx, rep);
+        RedisModule_FreeCallReply(rep);
+    }
+
+    return REDISMODULE_OK;
+}
+
+int rm_call_withclient_detached_context(RedisModuleCtx *ctx, RedisModuleString **argv, int argc){
+    if(argc < 2){
+        return RedisModule_WrongArity(ctx);
+    }
+
+    RedisModuleCtx *detached_ctx = RedisModule_GetDetachedThreadSafeContext(ctx);
```

Comment:
Lets use ThreadSafeContext

Reply:
Without blocking the client, isn't it the same (even should use detached)?

Reply:
Detached context should only be created at the module initialization. Even if it is the same it might break in the future so lets use it correctly.